### ISSUE TITLE
No selection list label in Finder component

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/joomla-cms#41198]
### Steps to reproduce the issue
1. Turn on a screen reader, such as NVDA
2. Run a search, for example, for the word 'joomla'
3. On the results page, set filters to specify the date range (selection options: 'before', 'after', 'exactly')

### Expected result
The screen reader announces the label of the option group "Start date", followed by the name of the first option "After".
Similarly, with the second option list "End date".
**Note*: This select list and comboboxes (with calendars) should have separate labels. The user hears that he has a list of choices, but does not know what the options are "After", "Before", "Exactly"   

### Actual result
The screen reader does not announce the option list label. These option groups have visible labels (visually related - 'Start date', 'End date'), but not programmatically related. The labels are linked programmatically to the comboboxes (select list+calendar) in which the user specifies dates.  

### Device and system info

* What version of Joomla were you using?
Joomla 4.3.3

* What device were you using? (eg, phone, laptop)
laptop

* What web browser were you using? (eg, Chrome, Safari)
Chrome

* If you were using assistive technology, please let us know the device or software you were using (eg VoiceOver, JAWS, sip-and-puff device, etc)
NVDA

### Additional comments

If you know which [WCAG](https://www.w3.org/TR/WCAG/) or [ATAG](https://www.w3.org/TR/ATAG20/) criteria this violates, please indicate it here:
SC 3.3.2, SC 2.5.3, SC 4.1.2

**Comment by brianteeman** [comment]
Note that to test this you need to go to the component options and set Date filters to show as they are hidden by default

**Comment by brianteeman** [comment]
And the relevant code to check/update ccan be found https://github.com/joomla/joomla-cms/blob/6a1a72ab25fd0722c9b4be2474a67dbe877d2585/administrator/components/com_finder/src/Service/HTML/Filter.php#L401
